Site Switcher: Introduce "Manage Sites" button at the bottom

zaguiini opened the pull request:
**IMPORTANT: This PR does NOT modify the "Add New Site" button inside the Site Switcher. That was addressed in https://github.com/Automattic/wp-calypso/pull/67346.**

#### Proposed Changes

Part of #67041. Let's add a new "Manage Sites" button at the bottom when opening the site switcher. This button will send the user to `/sites` once clicked. It will preserve the search query so they can reproduce that when browsing. A Tracks event `calypso_manage_sites_click` is dispatched whenever this button is clicked: 1045-gh-Automattic/tracks-events-registration.

Also, we now have proper multi-site context labeling:

![image](https://user-images.githubusercontent.com/26530524/188038351-32f8ff6b-56bc-4067-a267-2ee3b29e55fd.png)

The "All My Sites" label was very misleading and did not indicate that the user will see, for example, all posts for their sites. Now they do. I've updated the tracks event (1043-gh-Automattic/tracks-events-registration) to include the slug for the multi-site context page that the user is seeing.

### Testing Instructions

#### Manage sites button

1. Click `Switch Site`;
2. Check that "Manage Sites" button is stuck to the bottom of the page, even if you scroll the sites list:
![image](https://user-images.githubusercontent.com/26530524/188041183-658c248f-6b7a-4dbd-8f1d-54d72c4179b7.png)
3. Type any term on the search bar. Check that the number of sites reflects the search results count;
4. Check that the "Manage Sites" button preserves the search query and opens the Sites dashboard with the pre-filled term;
5. Check that a Tracks event called `calypso_manage_sites_click` was dispatched.

#### Multi-site context pages

1. Click `Posts` on any selected site;
2. Click `Switch Site`;
3. Check that the CTA is now aware of the all-sites contextual section:
![image](https://user-images.githubusercontent.com/26530524/188041443-e9bd2ec0-9db0-4758-b95f-897d9669528a.png)
4. Click that button;
6. Check that it's gone when opening the site switcher again. It should not show up when the user is already browsing such kind of page;
7. Check that a Tracks event called `calypso_all_my_sites_click` was dispatched with a `multi_site_context_slug` property.

#### Overall regression testing

Check that Jetpack Cloud (`yarn start-jetpack-cloud`) is unaffected by the changes, neither are`<SiteSelector />` usages across Calypso (e.g. http://calypso.localhost:3000/marketing/tools).

zaguiini committed:
Only show all sites page for items that have an all-sites version

zaguiini committed:
Add manage sites button at the bottom of the site selector

zaguiini committed:
Preserve search term when browsing to /sites

zaguiini committed:
Do not specify height for site-selector

zaguiini committed:
Move to showManageSites prop so we don't have to guess whether to show the button or not

zaguiini committed:
Match /posts and /posts/my when checking for an all-sites version for section

zaguiini committed:
Make manage sites button less proeminent when idle, and more proeminent on hover

zaguiini committed:
Do not display the count of sites to manage if its zero

zaguiini committed:
Clean up all-sites section CTA conditional

zaguiini committed:
Clarify multi-site context CTA variables and comments

zaguiini committed:
Add multi-site context slug to calypso_all_my_sites_click Tracks event

zaguiini committed:
Add manage sites click Tracks event

danielbachhuber commented on client/components/search-sites/index.js:
```diff
@@ -13,18 +13,19 @@ export default function searchSites( WrappedComponent ) {
 		getSearchResults() {
 			return this.state.term
-				? searchCollection( this.props.sites, this.state.term.toLowerCase(), [ 'name', 'URL' ] )
+				? searchCollection( this.props.sites, this.state.term, [ 'name', 'URL' ] )
```
@zaguiini Think we should add your fuzzy searching in a follow-up PR?

danielbachhuber replied:
Created #67316

danielbachhuber reviewed (dismissed):
Looks solid, nice work 🛠️ 

I've confirmed:
* The "Manage [N] Sites" button appears as expected. Clicking the button triggers the `calypso_manage_sites_click` event.
* The "View [feature] for all sites" button only appears when that feature is present for all sites.

Interestingly, the "Manage [N] Sites" button appears for me in Jetpack Cloud:

<img width="424" alt="image" src="https://user-images.githubusercontent.com/36432/188136559-13bbca93-846b-4ef8-9a27-4cad02d9bbe6.png">

I started Jetpack Cloud with `yarn start-jetpack-cloud-p`.

Also, this number discrepancy is a little odd:

<img width="404" alt="image" src="https://user-images.githubusercontent.com/36432/188138026-e162bc4d-ae26-4f3f-b247-0d55b9bac662.png">

Let's drop the count from the button and use "Manage Sites" instead.

Lastly, let's add the "Add new site" button to this PR, and then we can ship early next week.

zaguiini commented:
>Lastly, let's add the "Add new site" button to this PR, and then we can ship early next week.

I think I'll add it as a separate PR, trying to separate concerns here.

danielbachhuber commented:
> I think I'll add it as a separate PR, trying to separate concerns here.

Sounds good.

zaguiini committed:
Do not display the count of sites to manage at all

zaguiini committed:
Rename showManageSites to something more descriptive and add a prop type

zaguiini commented:
@danielbachhuber, changes made. Could you please retry?

I could not reproduce the `Manage Sites` button on the JC dashboard 😢. It does not happen if I run JC exclusively, neither if I run with Calypso side-by-side, nor in JC live Docker instance in this PR... Got me thinking. Can you still reproduce the problem?

zaguiini committed:
Do not extend the site list to the bottom of the page in Jetpack Dashboard

zaguiini committed:
Darken Manage Sites button on hover instead of making it lighter

p-jackson commented:
I think the `Manage sites` button needs to take better account of the current "interface theme".

This is what it looks like when the user has chosen the "Light" theme

<img width="308" alt="CleanShot 2022-09-06 at 15 08 47@2x" src="https://user-images.githubusercontent.com/1500769/188539215-ac152e78-0910-4c52-88d3-23c6d9e8d373.png">


I think there should be a "foreground colour" variable that we can use for the text and border colours.

p-jackson committed:
Button copies text and hover styles from the theme

p-jackson commented:
So I ended up copying the colours that the list items use. The "Manage sites" button sorta looks like a plain text button, and so having it use the same hover effect as the text looks good. Especially for this theme:

![Sep-06-2022 15-38-19](https://user-images.githubusercontent.com/1500769/188544304-c934870a-43e6-46e3-a604-1a01dad38e9c.gif)

It's like it's made for it. Unfortunately not every theme changes the text colour on hover. Some of them set a background colour. So I've done the same since I know each of those colours have been tested in the various themes.

![Sep-06-2022 15-37-44](https://user-images.githubusercontent.com/1500769/188544439-a03e5a7c-36c3-4510-a450-b13c537e8bf7.gif)
![Sep-06-2022 15-37-10](https://user-images.githubusercontent.com/1500769/188544456-c99c10f8-993d-4824-afc4-693261e80390.gif)

It doesn't look _too_ bad, although the "coloured background with white border" style might be a little unexpected...

I also tried something like `rgba( var( --color-sidebar-text-rgb ), 0.75 )` for the hover style so that we wouldn't have to have the filled-in background for the hover state. This works ok, but it means that for themes like the first GIF above the text hover colour for the button doesn't match the text hover colour for the list items. And that looks a bit peculiar.

p-jackson committed:
Don't show Manage sites button in JP Cloud

p-jackson committed:
Individual translations for each "all sites" nav label

p-jackson committed:
Remove unnecessary labels from Manage sites button

It's already an `<a>` tag with a visible text label, it doesn't require
any additional markup.

p-jackson committed:
Associate comment with the correct line of code

p-jackson commented on client/layout/index.jsx:
```diff
@@ -49,9 +49,12 @@ import { handleScroll } from './utils';
+	const active =
+		! isNarrow &&
+		! config.isEnabled( 'jetpack-cloud' ) && // Jetpack cloud hasn't yet aligned with WPCOM.
+		layoutFocus !== 'sites';
```
@danielbachhuber do you about why this `SidebarScrollSynchronizer` component needed to be deactivated when in the `sites` layout focus? I removed this line and everything worked as I expected.

danielbachhuber replied:
> do you about why this `SidebarScrollSynchronizer` component needed to be deactivated when in the `sites` layout focus? I removed this line and everything worked as I expected.

I don't, no. Everything works as expected when I remove it too.

I've reverted it with a1085d287b

p-jackson commented on client/my-sites/sidebar/static-data/all-sites-menu.js:
```diff
@@ -12,41 +12,47 @@ export default function allSitesMenu() {
 			icon: 'dashicons-chart-bar',
 			slug: 'stats',
 			title: translate( 'Stats' ),
+			navigationLabel: translate( 'View stats for all sites' ),
```
I've added individual translations for each of the "all sites" links. Previously we were using string interpolation and assuming it'd be ok to just make the `title` lowercase. I'm not sure but that seemed a bit English-centric. Maybe there's some gendered-ness or something else that'll mean the title can't just be inserted in the middle of a sentence.

p-jackson reviewed (commented):
I was able to see the "Manage site" button in the Jetpack Cloud switcher too, so I've fixed that by exposing the `showManageSitesButton` prop further up the component tree.

I made some other changes like removing unnecessary aria labels (the button is already accessible enough to screen readers).

I think this is good to go now, although I did leave a comment about `SidebarScrollSynchronizer` below because I don't know why that change was necessary.

danielbachhuber committed:
Merge branch 'trunk' into add/manage-sites-button-site-switcher

danielbachhuber committed:
Partial revert of 537c6dae9fa7eed78ca814a3105aa0f38fa847c2

danielbachhuber committed:
Ensure the style override applies to all of Jetpack Cloud

danielbachhuber committed:
Merge branch 'trunk' into add/manage-sites-button-site-switcher

danielbachhuber reviewed (approved):
Looks good 👍 

I tested in Chrome, Firefox, and Safari on Mac. I also tested with a variety of dashboard color schemes.

The 'Add new site' button ended up stuck to the bottom in non-agency JP Cloud views, so I broadened the CSS override with 9ce12232cb5f80bf02da095e5a953ea1111f2007

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7513767

Thank you @zaguiini for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7513767).
